data/spells/scripts/support/disintegrate rune.lua

```diff
@@ -0,0 +1,24 @@
+local corpseIds = {3058, 3059, 3060, 3061, 3064, 3065, 3066}
+local removalLimit = 500
+
+function onCastSpell(creature, variant, isHotkey)
+	local position = variant:getPosition()
+	local tile = Tile(position)
+	if tile then
+		local items = tile:getItems()
+		if items then
+			for i, item in ipairs(items) do
+				if not isInArray(corpseIds, item:getId()) and item:getType():isMovable() and item:getUniqueId() > 65535 and item:getActionId() == 0 then
```

Comment:
I think `isInArray` is the most expensive check and could be moved to the end of the comparison.